lib/MessageImpl.h

```diff
@@ -47,6 +47,7 @@ class MessageImpl {
     int redeliveryCount_;
     bool hasSchemaVersion_;
     const std::string* schemaVersion_;
+    std::shared_ptr<class ConsumerImpl> consumerPtr_;
```

Comment:
Should you store a `std::weak_ptr`? If we have N `Message` objects, the reference count of the `ConsumerImpl` will be at least `N + 1`, which is unnecessary and it might extend the lifetime of internal consumers unexpectedly.